### ISSUE TITLE
Add a transformer to jest

### ISSUE THREAD

**Issue by ekaradon** [timarney/react-app-rewired#210]
Hello,

I am already using this project in order to rewire `eslint` as I want. 
However, I would like now to add a transformer when running tests. 
And so I changed `package.json` with the following content:

```
  "jest": {
    "transform": {
      "\\.(js)$": "<rootDir>/node_modules/react-i18nliner/jest-transform.js"
    }
  },
```
But then I got the following output:

```
 FAIL  src/**/*.test.js
  ● Test suite failed to run

    src/setupTests.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import jasmine from 'jasmine';
```

With the following packages:
 -    "react-app-rewired": "1.4.0"
 -    "react-i18nliner": "0.2.3"

With a temporary custom file within `react-i18nliner` containing the following:
```
var through = require("through2");
var I18nliner = require("i18nliner").default;
var config = I18nliner.config;
var preprocess = require("./preprocess");
var hasTranslatableText = require("./hasTranslatableText")(config);

module.exports = {
    process(src) {
        if (hasTranslatableText(src)) {
            return src = preprocess.call({cacheable() {}}, src, config);
        }
        return src;
    }
};
rce;
};
```
After investigating, I found that it was called before `babel` transformers... so I used the dynamic override in order to ensure to put my transformer at the end. 

```
const rewireEslint = require('react-app-rewire-eslint');

/* config-overrides.js */
module.exports = {
  webpack: (config, env) => {
    return rewireEslint(config, env);
  }
  , jest: (config, env) => {
    config.transform['^.+\\.js$'] = `${config.rootDir}/node_modules/react-i18nliner/jest-transform.js`;
    return config;
  }
}
```

And now it works... except the transformer is never called and I really don't see why. I really could use some help here, what am I missing here?

Thanks

**Comment by timarney** [comment]
@dawnmist (sorry another jest one).  Any thoughts.

**Comment by dawnmist** [comment]
@ekaradon The first thing I'd suggest double-checking is that you're overwriting the original javascript transformer rather than defining a second one that then is ignored because your javascript files had already matched one of the existing ones. I know that when I was trying to get this working with our typescript environment that it took a bit of fiddling to get it to work properly. I ended up copying the parts of the react-scripts-ts jest transformers we wanted to keep into our own definitions, and simply replacing the entire transform variable. It meant that if they adjusted the file-matcher key later our definitions didn't suddenly break - but it also meant that if they added new transforms we wouldn't automatically also get the update until I copied the new section across to our definitions.

Assuming that you are using current create-react-app's react-scripts package (i.e. assuming you're not using the `--scripts-version` option in react-app-rewired), I think the transform key that you need to be overwriting is probably `'^.+\\.(js|jsx|mjs)$'`. https://github.com/facebook/create-react-app/blob/1e52f252fcc153dc4cfc78f658d2ac5fac4ec3e7/packages/react-scripts/scripts/utils/createJestConfig.js#L38-L45

**Comment by ekaradon** [comment]
Hello,

Thanks for your help.
I have already understood that after some times yesterday. And I tried to overwrite it by just doing what you say. 
And I have done something like `babel_plugins.process(react-i18nliner(src));`. 
However I got some really strange behaviour. The tests were runned without having the actual source being transpiled (or I got some really huge time difference between output of test files and source files). And ultimately, I still got the warnings complaining about the presence of the `translate` property. 
It triggered me and after seeing that `react-i18nliner` was not updated since almost one year, I just decided to get rid of it. 

So I close this issue. However it really look like something a bit painful to do, not being able to use several transformers for a same file.

Anyway, thanks for your help!
